src/App.tsx

```diff
@@ -1,204 +1,119 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState, useEffect } from 'react';
+import { getTodos } from './api/todos';
 import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import TodoForm from './components/TodoForm';
+import TodoList from './components/TodoList';
+import Footer from './components/Footer';
+import ErrorNotification from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState('all');
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      setLoading(true);
+      setError(null);
+
+      try {
+        const todosFromApi = await getTodos();
+
+        setTodos(todosFromApi);
+      } catch {
+        setError('Unable to load todos');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (USER_ID) {
+      loadTodos();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      const timer = setTimeout(() => {
+        setError(null);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [error]);
+
+  const handleAddTodo = (title: string) => {
+    if (title.trim()) {
+      setTodos([
+        ...todos,
+        {
+          id: Date.now(),
+          title: title.trim(),
+          completed: false,
+          userId: 0,
+        },
+      ]);
+    }
+  };
+
+  const handleToggleAll = () => {
+    const allCompleted = todos.every(todo => todo.completed);
+
+    setTodos(todos.map(todo => ({ ...todo, completed: !allCompleted })));
+  };
+
+  const handleFilterChange = (newFilter: string) => setFilter(newFilter);
+
+  const handleClearCompleted = () => {
+    setTodos(todos.filter(todo => !todo.completed));
+  };
+
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'active') {
+      return !todo.completed;
+    }
+
+    if (filter === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
 
   return (
     <div className="todoapp">
-      <h1 className="todoapp__title">todos</h1>
-
+      <h1 className="todoapp__title" data-cy="TodoAppTitle">
+        todos
+      </h1>
       <div className="todoapp__content">
         <header className="todoapp__header">
-          {/* this button should have `active` class only if all todos are completed */}
           <button
             type="button"
-            className="todoapp__toggle-all active"
+            aria-label="Mark all as completed"
+            className={`todoapp__toggle-all ${todos.every(todo => todo.completed) ? 'active' : ''}`}
```

Comment:
Use the classnames library for add classes with condition, fix it everywhere

Reply:
done in App.tsx, Footer.tsx, TodoItem.tsx